Prox/Prox/Widgets/ReviewScoreView.swift

```diff
@@ -62,34 +57,58 @@ class ReviewScoreView: UIStackView {
         }
     }
 
+    private func updateReviewScore(score: Float) {
+        for (index, subview) in arrangedSubviews.enumerated() {
+            if let scoreItem = subview as? ReviewScoreItemView {
+                let scoreLeft = self.score - Float(index)
+                setFillScore(score: scoreLeft, forView: scoreItem)
+            }
+        }
+    }
+
+    private func setFillScore(score: Float, forView view: ReviewScoreItemView) {
+        if score >= 1 {
+            view.leftColor = color
+            view.rightColor = color
+        } else if score > 0 {
+            view.leftColor = color
+            view.rightColor = Colors.reviewScoreDefault
+        } else {
+            view.leftColor = Colors.reviewScoreDefault
+            view.rightColor = Colors.reviewScoreDefault
+        }
+    }
+
 }
 
 private class ReviewScoreItemView: UIView {
 
-    var color: UIColor? {
+    var leftColor: UIColor? {
         didSet {
             setNeedsDisplay()
         }
     }
-    var fill: Bool = false {
+
+    var rightColor: UIColor? {
         didSet {
-            if fill != oldValue {
-                setNeedsDisplay()
-            }
+            setNeedsDisplay()
         }
     }
 
     override func draw(_ rect: CGRect) {
         super.draw(rect)
 
-        if let context = UIGraphicsGetCurrentContext() {
-            if fill {
-                context.setFillColor(color?.cgColor ?? UIColor.black.cgColor)
-            } else {
-                context.setFillColor(Colors.reviewScoreDefault.cgColor)
-            }
-            context.fillEllipse(in: rect)
-        }
+        let width = rect.size.width / 2
+
+        let leftHalf = UIBezierPath(roundedRect: CGRect(x: rect.origin.x, y: rect.origin.x, width: width, height: rect.size.height), byRoundingCorners: [UIRectCorner.topLeft, UIRectCorner.bottomLeft], cornerRadii: CGSize(width: 10, height: 10))
+        leftHalf.close()
+        (leftColor ?? UIColor.black).setFill()
```

Comment:
> It was just so that we had a default color in case no one set a color

I prefer to crash when the app is in an unexpected state – assuming you can get crash reports, you can find errors early and fix them, rather than having an error which may be a slightly less bad experience but that you may never fix. This is particularly important when errors may compound (e.g. one view being black is alright but when all the views on the screen are black).

Thinking about it, it's less important for UI but more so for data processing – which is perhaps similar to your point about ViewControllers throwing the fit rather than the views.